### DISCUSSION TITLE
Do we loose intellisense in Mantine 7?

### DISCUSSION THREAD

**Discussion by branislavbrincko** [mantinedev/mantine#4754] [Q&A]
I just stumbled upon the beta version of Mantine 7 where the most prominent change is switch from Emotion to CSS modules leveraging  --mantine- CSS variables.
One thing that bothers me the most - do we now loose intellisense/typesafety when working with theme values? What I mean:
In the Mantine 6 I can do something like this: 
<img width="573" alt="image" src="https://github.com/mantinedev/mantine/assets/56249098/b7392fe3-6408-4a12-84b5-34d78014b524">
and I get the intellisense/typesafety (simply I cannot use the non existent color). 
But now, using css variables, it can easily happen: 
<img width="735" alt="image" src="https://github.com/mantinedev/mantine/assets/56249098/56e4cdf6-0f3e-4175-9862-3000c81d80a1">
I get no warning that I am using non existent color. 
This is probably non Mantine specific question, but is there a way to have some kind of validation for project-available CSS variables? Is there a recommended way to have intellisense when accessing --mantine variables?

**Comment by rtivital** [comment]
If you prefer to continue using ts for styles, see https://v7.mantine.dev/styles/vanilla-extract
To add autocomplete for variables install extension for your editor, for example https://marketplace.visualstudio.com/items?itemName=vunguyentuan.vscode-css-variables
If you want to avoid typos install https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

**Reply by Tsourdox** [reply]
Thank you that was exactly what i was looking for. Is there a way to also get intellisense for the new PostCSS mixins and functions?

**Reply by rtivital** [reply]
I do not think so

**Reply by ScottKemp** [reply]
I could not get the auto complete of the variables to work with the extension installed until I added
{
  "cssVariables.lookupFiles": [
    "**/*.css",
    "**/*.scss",
    "**/*.sass",
    "**/*.less",
    "node_modules/@mantine/core/styles/global.css"
  ]
}

To my .vscode/settings.json file

**Reply by rtivital** [reply]
Setup for VS Code is described on the getting started page – https://mantine.dev/getting-started/#set-up-vs-code

**Reply by green-0-rabbit** [reply]
In case someone else find this thread, you could install [postcss by vunguyentuan](https://marketplace.visualstudio.com/items?itemName=vunguyentuan.vscode-postcss) to highlight @mixin and other postcss syntax
